core-processor/src/ext.rs

```diff
@@ -841,14 +849,37 @@ impl<LP: LazyPagesInterface> CountersOwner for Ext<LP> {
             CounterType::GasLimit => gas.checked_sub(amount),
             CounterType::GasAllowance => allowance.checked_sub(amount),
         }
-        .unwrap_or_else(|| unreachable!("Checked above"));
+        .unwrap_or_else(|| {
+            let err_msg = format!(
+                "CounterOwner::decrease_current_counter_to: Checked sub operation overflowed. \
+                Message id - {}, current counter type - {:?}, gas - {gas}, allowance - {allowance}, amount - {amount}",
```

Comment:
if message id is state, I'd state the program id too

Reply:
same below

Reply:
Fixed